spec/mongoid/contextual/mongo_spec.rb

```diff
@@ -195,6 +195,52 @@
     end
   end
 
+  describe "#estimated_count" do
+
+    let!(:depeche) do
+      Band.create(name: "Depeche Mode")
+    end
+
+    let!(:new_order) do
+      Band.create(name: "New Order")
+    end
+
+    let(:criteria) do
+      Band.where
+    end
+
+    context "when not providing options" do 
+      it 'returns the correct count' do 
+        expect(criteria.estimated_count).to eq(2)
+      end
+    end
+
+    context "when providing options" do 
+      it 'returns the correct count' do 
+        expect(criteria.estimated_count(maxTimeMS: 1000)).to eq(2)
+      end
+    end
+
+    context "when context is cached" do
+
+      let(:context) do
+        described_class.new(criteria.cache)
+      end
+
+      before do
+        expect(context.view).to receive(:estimated_document_count).once.and_return(1)
+      end
+
+      it "returns the count cached value after first call" do
+        2.times do 
+          context.estimated_count
```

Comment:
This test has no assertions?

Reply:
Please move the expectation setup out of before into the example so that it is clear that there are assertions.

Reply:
I found a similar pattern in other places, I fixed them as well.